source/agora/consensus/protocol/Nominator.d

```diff
@@ -860,9 +870,14 @@ extern(D):
 
             Hash random_seed = this.ledger.getExternalizedRandomSeed(
                 this.ledger.getBlockHeight(), data.missing_validators);
+
+            Transaction[] externalized_tx_set;
+            if (this.ledger.getValidTXSet(data, externalized_tx_set))
+                return;
```

Comment:
I feel like this should be an assert(0) here as we can not ignore the invalid tx set when we should be externalizing the block.

Reply:
Since we now externalize TX hashes, we might not have all the TXs in our pool at the point the value was externalized. So that could fail for missing TXs as well. In that case we should be able to catch up to the network and sign the unsigned blocks along the way.

Reply:
So will this externalize be triggered again after the missing TXs are received?

Reply:
No, node will have to download the full block from the network.

Reply:
Perhaps I should also not assert in that case, when there are not enough signatures. The node can also wait to recieve the block during a periodic catchup and test again for required number of signatures before externalizing.

Reply:
I created [PR #1517](https://github.com/bpfkorea/agora/pull/1517) to address this for block sigs. I agree with your logic here.

Reply:
Great, thanks!